### PR TITLE
Bump Dart SDK to `1.20.0-dev.10.0`.

### PR DESCRIPTION
**Pull request by pq**
`1.20.0-dev.10.0` corresponds to `1.20-RC0`.

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

**Comment by pq** [comment]
Thanks @abarth !

Before this lands I need another set of eyes as I added a bit of extra logic to the `flutter` script to do some necessary cleanup.

Net-net, upgrading an existing install fails because of a stale `pubspec.lock` in `flutter_tools`.  (It turns out lock files don't play well with dependency overrides...)

cc @devoncarew

File: bin/flutter [hunk 1]
```diff
@@ -41,7 +41,7 @@ DART="$DART_SDK_PATH/bin/dart"
 
     echo Building flutter tool...
     FLUTTER_DIR="$FLUTTER_ROOT/packages/flutter"
-    (cd "$FLUTTER_TOOLS_DIR"; "../../bin/cache/dart-sdk/bin/pub" get --verbosity=error --no-packages-dir)
+    (cd "$FLUTTER_TOOLS_DIR"; if test -f pubspec.lock; then rm pubspec.lock; fi; "../../bin/cache/dart-sdk/bin/pub" get --verbosity=error --no-packages-dir)
```
Comment (devoncarew) [inline bin/flutter]: This guy could be simplified a bit to just `rf -f pubspec.lock`.

**Comment by devoncarew** [comment]
lgtm

File: bin/flutter (hunk 1 above)
Comment (pq) [inline bin/flutter]: Ah, `rm -f`.  Right!

**Pull request merged**